### PR TITLE
[IOTDB-1239]Incorrect datetime value can be inserted into iotdb succe…

### PR DESCRIPTION
**Pull request by lyssom**
…ssfully.

## Description
fix IOTDB-1239
https://issues.apache.org/jira/browse/IOTDB-1239
support big datetime parse and query

### Content1 ...
result like this ...

IoTDB> SET STORAGE GROUP TO root.sg
Msg: The statement is executed successfully.
IoTDB> CREATE TIMESERIES root.sg.d1.s2 WITH DATATYPE=DOUBLE, ENCODING=PLAIN
insert into root.sg.d1(time,s2 ) values (1574179200000000,8.76);
Msg: The statement is executed successfully.
IoTDB> 
Msg: The statement is executed successfully.
IoTDB> select * from root.sg.d1;
+-------------------------------+-------------+
|                           Time|root.sg.d1.s2|
+-------------------------------+-------------+
|+51853-10-03T00:00:00.000+08:00|         8.76|
+-------------------------------+-------------+
Total line number = 1
It costs 0.328s
IoTDB> select * from root.sg.d1 where time=51853-10-03T00:00:00.000+08:00
+-------------------------------+-------------+
|                           Time|root.sg.d1.s2|
+-------------------------------+-------------+
|+51853-10-03T00:00:00.000+08:00|         8.76|
+-------------------------------+-------------+
Total line number = 1
It costs 0.049s
IoTDB> 
### Content3 ...

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [x] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: service-rpc/src/main/java/org/apache/iotdb/rpc/RpcUtils.java [hunk 1]
```diff
@@ -244,7 +244,8 @@ public static String parseLongToDateWithPrecision(
           digits.insert(0, "0");
         }
       }
-      return datetime.substring(0, 19) + "." + digits + datetime.substring(19);
+      String timeZoneStr = datetime.substring(datetime.length() - 6);
+      return datetime.substring(0, datetime.length() - 6) + "." + digits + timeZoneStr;
```
Comment (HTHou) [inline service-rpc/src/main/java/org/apache/iotdb/rpc/RpcUtils.java:248]: Hi, this PR looks good to me. 
I think we should also be careful when the timestampPrecision is us or ns.
Also, please check the failed tests. Thank you so much!

Comment (lyssom) [reply service-rpc/src/main/java/org/apache/iotdb/rpc/RpcUtils.java:248]: ok，

**Comment by HTHou** [comment]
Hi, some tests related with Zeppelin connector failed. Can you please take a look?

File: service-rpc/src/main/java/org/apache/iotdb/rpc/RpcUtils.java (hunk 1 above)
Comment (HTHou) [inline service-rpc/src/main/java/org/apache/iotdb/rpc/RpcUtils.java:248]:
```suggestion
      String timeZoneStr = datetime.substring(datetime.indexOf('Z'));
      return datetime.substring(0, datetime.indexOf('Z')) + "." + digits + timeZoneStr;
```

File: service-rpc/src/main/java/org/apache/iotdb/rpc/RpcUtils.java [hunk 2]
```diff
@@ -257,7 +258,8 @@ public static String parseLongToDateWithPrecision(
...
```
Comment (HTHou) [inline service-rpc/src/main/java/org/apache/iotdb/rpc/RpcUtils.java:262]:
```suggestion
      String timeZoneStr = datetime.substring(datetime.indexOf('Z'));
      return datetime.substring(0, datetime.indexOf('Z')) + "." + digits + timeZoneStr;
```

File: service-rpc/src/main/java/org/apache/iotdb/rpc/RpcUtils.java [hunk 3]
```diff
@@ -270,7 +272,8 @@ public static String parseLongToDateWithPrecision(
...
```
Comment (HTHou) [inline service-rpc/src/main/java/org/apache/iotdb/rpc/RpcUtils.java:276]:
```suggestion
      String timeZoneStr = datetime.substring(datetime.indexOf('Z'));
      return datetime.substring(0, datetime.indexOf('Z')) + "." + digits + timeZoneStr;
```

**Review by HTHou — CHANGES_REQUESTED** [review]

**Comment by lyssom** [comment]
> 
> 
> Hi, some tests related with Zeppelin connector failed. Can you please take a look?

ok， I fixed it

**Comment by HTHou** [comment]
What is .metals/metals.h2.db?

**Comment by lyssom** [comment]
> What is .metals/metals.h2.db?
> Screen Shot 2021-04-29 at 5 44 20 PM

sorry，it vs code update file.this file is wrong to commit。i will delete it

**Comment by HTHou** [comment]
By the way, some one mentioned another problem about query, could you please take a look?  
https://issues.apache.org/jira/browse/IOTDB-1239

**Comment by lyssom** [comment]
> By the way, someone mentioned another problem about query, could you please take a look?
> https://issues.apache.org/jira/browse/IOTDB-1239

ok，，it fixed. In Datetimeutiles, I changed datefomatbuilder max value 10 to 19 to solve it.

**Review by HTHou — APPROVED** [review]
LGTM

**Comment by HTHou** [comment]
> > By the way, someone mentioned another problem about query, could you please take a look?
> > https://issues.apache.org/jira/browse/IOTDB-1239
> 
> ok，，it fixed. In Datetimeutiles, I changed datefomatbuilder max value 10 to 19 to solve it.

Should we need to change this part as well?

https://github.com/apache/iotdb/blob/d97cd67fb02ca6823e14344074f7d6b2be43b6bd/server/src/main/java/org/apache/iotdb/db/qp/utils/DatetimeUtils.java#L55-L80

**Comment by lyssom** [comment]
> > > By the way, someone mentioned another problem about query, could you please take a look?
> > > https://issues.apache.org/jira/browse/IOTDB-1239
> > 
> > 
> > ok，，it fixed. In Datetimeutiles, I changed datefomatbuilder max value 10 to 19 to solve it.
> 
> Should we need to change this part as well?
> 
> https://github.com/apache/iotdb/blob/d97cd67fb02ca6823e14344074f7d6b2be43b6bd/server/src/main/java/org/apache/iotdb/db/qp/utils/DatetimeUtils.java#L55-L80

yes，I miss it. These formats should be support.I will change them.

**Pull request closed without merging**